src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -4006,6 +4008,14 @@ void EmitLoop(RegexNode node)
                 {
                     doneLabel = originalDoneLabel;
                     MarkLabel(endLoop);
+
+                    // The loop is atomic, which means any backtracking will go around this loop.  That also means we can't leave
+                    // stack polluted with state from successful iterations, so we need to remove all such state; such state will
+                    // only have been pushed if minIterations > 0.
+                    if (startingStackpos is not null)
+                    {
+                        writer.WriteLine($"stackpos = {startingStackpos};");
```

Comment:
Nit: I wonder  if this is one of those lines in the generated source that might warrant a small comment above explaining the assignment for readability/comprehension when debugging.